hpx/components/containers/partitioned_vector/partitioned_vector_segmented_iterator.hpp

```diff
@@ -521,6 +521,11 @@ namespace hpx
           : base_type(it), data_(data)
         {}
 
+        operator BaseIter const &() const
```

Comment:
`segmented_iterator_traits` needs to know the template parameter `BaseIter` of `segment_vector_iterator` to _explicitly_ cast the input segment_vector_iterator inside the newly implemented `local_segment()` method. But there's neither public `segment_vector_iterator`'s inner typedef  (think it was intentional), nor `segmented_iterator_traits`'s inner typedef corresponding to this `BaseIter`